src/event-bridge/transform.ts

```diff
@@ -59,19 +58,21 @@ export class EventBusTransform<T extends EventBusRuleInput, P> {
    *
    * @see EventBusRule.pipe for more details on pipe.
    */
-  pipe(props: LambdaTargetProps<P>): void;
-  pipe(func: IFunction<P, any>): void;
-  pipe(props: StateMachineTargetProps<P>): void;
-  pipe(props: StepFunction<P, any>): void;
-  pipe(props: ExpressStepFunction<P, any>): void;
-  pipe(
-    resource:
-      | IFunction<P, any>
-      | LambdaTargetProps<P>
-      | StateMachineTargetProps<P>
-      | StepFunction<P, any>
-      | ExpressStepFunction<P, any>
-  ): void {
-    pipe(this.rule, resource as any, this.targetInput);
+  pipe<
+    I extends Integration<any, string, EventBusTargetIntegration<P, Props>> & {
+      eventBus: EventBusTargetIntegration<P, Props>;
+    },
+    Props extends object | undefined
+  >(
+    integration: NonEventBusIntegration<I>,
+    ...props: Parameters<DynamicProps<Props>>
+  ) {
+    pipe(this.rule, integration, props[0] as Props, this.targetInput);
```

Comment:
Why is this not returned?

Reply:
It can be, though it doesn't return a wrapped type.

Reply:
Curious for we will configure alarms?

Reply:
soo... there are no monitoring helpers on the event bus CDK constructs.

The metrics on event bus are not great, mostly limited to the bus or rule level.

https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-monitoring.html

So right now monitoring would be on:
1. The source bus or rule
2. The target
3. the dead letter queue

Reply:
Are these exposed on the return type of `.pipe`?

Reply:
`pipe` doesn't return anything right now because I didn't have a use case for it yet.

We could return:
1. the transform (`.pipe().pipe().rule`)
2. the rule (`.pipe().map().pipe()`)
3. the CDK IRuleTarget
4. a new Target object (what would this have?)
5. nothing

I guess the transform makes the most sense. Pipe itself doesn't really generate a resource, it adds a target to a rule and there is nothing you could really do with that target except for add it to another rule (which wouldn't be type safe and targets are cheap; aka not much value in reuse).  

I think I'll leave this for now as I don't see much utility in any of the use cases.